### ISSUE TITLE
Config backup contains password

### ISSUE THREAD

**Issue by mbeck-** [opnsense/core#3667]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

**Describe the bug**
Config backup contains root password in clear text under opnsense->notifications->growl->password.

Never used Growl notifications in OPNsense (seems not to be supported anymore anyway).

**To Reproduce**
1. Go to "Configuration" -> "Backup" (diag_backup.php)
2. Click "Download configuration" (no RRD data, no encryption)

**Expected behavior**
No passwords in clear text in configuration, especially in a section where it doesn't make sense.

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 19.7.2-amd64

Currently installed plugins:

os-dyndns, os-telegraf, os-api-backup

**Comment by fabianfrz** [comment]
If growl is not supported anymore, you can delete the unused node manually. Unused nodes are not removed automatically and stay in the configuration.

**Comment by fichtner** [comment]
Well, first, one needs to assess how the password was being filled. Worst case it was a browser auto fill.

Secondly, if the underlying remote service doesn’t support key-based authentication plain text passwords are necessary.

The same argument could be made for private SSL keys. It’s impractical to assume we can’t store this data.

Solution: encrypt your backups with a password.

**Comment by mbeck-** [comment]
> Well, first, one needs to assess how the password was being filled. Worst case it was a browser auto fill. Secondly, if the underlying remote service doesn’t support key-based authentication plain text passwords are necessary. The same argument could be made for private SSL keys. It’s impractical to assume we can’t store this data. Solution: encrypt your backups with a password.
>

Yes, I thought of browser auto fill, too. But I concluded that it's probably a bug since I can't find the password anywhere in the Web UI and I don't usually let the browser save passwords I care about (and I checked my browser: no password saved regarding OPNsense, at least as of now). I didn't know that unused nodes are not remove automatically, otherwise I would have tried to remove it manually, re-upload the configuration and try again. I'm going to try that. 

What do mean with "plain text passwords are necessary"? My regular root password in the config backup is hashed, which is what I expected.

Regarding encrypting the config file backup: I want to put it in version control as I do with everything else. Of course the configuration file contains sensitive information, but I don't want anyone passing me seeing a plain text root password.

As said, browser auto fill seems possible, trying to remove this node manually...
